scripts/get_latest_build_number.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -eu -o pipefail
+
+print_all_remote_refs () {
+  git for-each-ref --format='%(refname:lstrip=2)' refs/remotes | grep -vE '/HEAD$'
+}
+
+BUILD_NUMBERS=""
+for REMOTE_BRANCH in $(print_all_remote_refs); do
```

Comment:
I wonder if there is any potential issue here due to too many branches or some kind of pagination of the commands. But 0/5.

Reply:
I wouldn't expect pagination within noninteractive shells (there's the `--count` option for that command that could potentially limit its output, but it's not used).

Since we're not checking out the branches, performance shouldn't be a relevant issue I think. This currently takes ~0.2s on average on my machine, with the currently existing 127 branches. E.g.:
```
$ git for-each-ref --format='%(refname:lstrip=2)' refs/remotes | grep -vE '/HEAD$' | wc -l
127
$ time ./get_latest_build_number.sh 
495

real	0m0.211s
user	0m0.349s
sys	0m0.093s
```
There's room for improvement but at the cost of some readability, so I'd be in favour of keeping it this way, but i'm also 0/5 on this and wouldn't mind it either way

Reply:
There is a fastlane action you can use to get the build number (always use the main branch I guess) no need for this file and no need to iterate over branches

Reply:
My understanding is that sometimes the largest build number is on a branch that is not `main` (e.g. when there is a release or an RC release), in which case this automated weekly PR would be incorrect, since the fastlane action only looks at its own branch to figure out the next build number. If this happens and we don't manually check the `release-*` branches before approving the PR, the corresponding beta build would fail to push to the app stores, since the same build number was already used by another branch. Is this correct?
Ideally i'd like these recurring tasks to be as smooth as possible

Reply:
The build number in master ahould always be incremented whenever there is a new build.

The version number is a different matter

Reply:
Does it include release branches that are not the latest? E.g. should i push a new build number to master if i want to cherry-pick it to `release-2.10`, but the `release-2.11` branch already exists?

Reply:
During testing it turned out that currently we are forced to specify `BUILD_NUMBER` externally, for some reason: https://github.com/mattermost/delivery-platform/actions/runs/7118359479/job/19381136786#step:5:265
I'll re-add the script for now, but perhaps this could be handled by the fastfile itself in the future